gufe/tests/test_chemicalsystem.py

```diff
@@ -132,6 +133,39 @@ def test_sorting(solvated_complex, solvated_ligand):
     assert sorted(order1) == sorted(order2)
 
 
+def test_contains_wrong_type(solvated_complex):
+    with pytest.raises(TypeError, match="`item` must be an instance or subclass of `Component`"):
+        solvated_complex.contains(float)
+
+
+def test_contains_instance(solvated_complex, prot_comp, toluene_ligand_comp, phenol_ligand_comp):
+    # check for present instances don't return matches
+    assert solvated_complex.contains(prot_comp) is True
+    assert solvated_complex.contains(phenol_ligand_comp) is False
+
+
+def test_contains_type(solvated_complex):
+    # check for present types don't return matches
+    assert solvated_complex.contains(ProteinComponent) is True
+
+
+def test_get_components_of_type(solvated_ligand):
+    # check for present types
+    matches = solvated_ligand.get_components_of_type(ProteinComponent)
+    assert matches == []
+
+    matches = solvated_ligand.get_components_of_type(SmallMoleculeComponent)
+    assert matches == [solvated_ligand.components["ligand"]]
+
+    matches = solvated_ligand.get_components_of_type(SolventComponent)
```

Comment:
I'd like the sanity-check that `solvated_ligand.get_components_of_type(Component)` returns all components.